plugin/cmd/kube-scheduler/app/server.go

```diff
@@ -114,7 +114,7 @@ func Run(s *options.SchedulerServer) error {
 	rl := &resourcelock.EndpointsLock{
 		EndpointsMeta: metav1.ObjectMeta{
 			Namespace: "kube-system",
-			Name:      "kube-scheduler",
+			Name:      "kube-" + s.SchedulerName,
```

Comment:
this also means an old a new version of the scheduler can both acquire a lock simultaneously